### ISSUE TITLE
UX for kicking balls

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/center-and-variability#72]
Over in https://github.com/phetsims/geometric-optics/issues/355, I was asked to take a look at Center and Spread for a specific feature.  While I was doing that....

I really really want to click on the PhET kids, or at least the soccer ball next to them, to kick the ball.  I realize that there are 2 giant "Kick" buttons just below the kids. But I click on the kids every time.  And the buttons are just not fun, and so indirect. The buttons might be more fun/inviting with an icon on them instead of text, but they'd still be indirect.

**Comment by amanda-phet** [comment]
Thanks for sharing @pixelzoom . We've had a number of discussions about this and have wanted to do exactly what you are trying to do, but there have been some other usability concerns that have come up. If someone can touch/click the person/ball, will they think they can then drag the ball to the grass? The balls are draggable once they've landed, but not before that. I'm also curious what you thought after the first kick, when the ball has lower opacity. Did you still want to click the player/ball then? Or just for the first kick?

**Comment by pixelzoom** [comment]
It was honestly quite awhile before I noticed that the kickable ball had become translucent -- only after I tried to verbalize (for this issue) what I wanted to be able to do.  So I don't think transparency is effective in any case.   It certainly had no affect on my instinctive actions.

I also think that you're also potentially causing a usability problem by not _all_ balls (including the kickable ball) interactive.  Because I couldn't interact directly with the kickable ball, I never tried to move the balls that had been kicked -- the exact opposite of what you're trying to achieve.  Users are more likely to interact with the balls that have been kicked if they can interact (albeit in a different way) with the back to be kicked.

**Comment by samreid** [comment]
Clicking on a soccer player or unkicked ball to kick it sounds reasonable and not too difficult to implement.

**Comment by amanda-phet** [comment]
If we have time, I'd like to add this to the classroom prototype too!

**Comment by samreid** [comment]
From Slack:

> Do you want https://github.com/phetsims/center-and-variability/issues/72 to trigger on mouse down or mouse up?

@amanda-phet replied:
> Mouse down. I remember discussing that so that users didn’t thin[k] they could drag the ball

**Comment by samreid** [comment]
More from slack:


Sam Reid  11:48 AM
Do you want https://github.com/phetsims/center-and-variability/issues/72 to trigger on mouse down or mouse up?






Amanda McGarry  11:49 AM
Mouse down
11:49
I remember discussing that so that users didn’t thin they could drag the ball

Sam Reid  11:56 AM
Do we want fire on hold for that?

Amanda McGarry  11:59 AM
Maybe? Can we control how quickly it fires?

Sam Reid  12:00 PM
Yes
12:00
I presume we would match it to the rate of “fire on hold” for the buttons?

Amanda McGarry  12:00 PM
OK, then yes, and match.. yes, that’s what I was going to say :slightly_smiling_face:

Sam Reid  12:00 PM
But we would implement it differently, since the first soccer player disappears after they kick.
12:00
OK we will give it a try.
New

Amanda McGarry  12:00 PM
sounds good, and if it’s becoming complicated, we can skip fire on hold for now

Sam Reid  12:02 PM
This seems like 1-3 hours for this feature with fire on hold.  Or maybe 1-1.5 hours to implement “click to kick” without fire on hold. (edited) 

Amanda McGarry  12:03 PM
Perhaps we should not fit this in for the prototype, since we need a dev version ready today.
12:03
Want me to jump on the zoom call to discuss?

Sam Reid  12:04 PM
  Perhaps we should not fit this in for the prototype, since we need a dev version ready today.
That decision sounds reasonable, but we are still on the call and wrapping up if you want to touch base.

**Comment by samreid** [comment]
In discussion with @amanda-phet and @chrisklus, we agreed there isn't time to do this feature in time for the milestone, so we will remove it from the milestone.  That will also give us more time to decide on the details like press to hold, and whether to use one input listener to cover the player and ball, or something else.

**Comment by samreid** [comment]
From today's design meeting:

* If we make it so you can click on the soccer player or the soccer ball, it would behave like the kick one button. Trigger on mouse down, and "fire on hold".
* Don't need soccer player or ball in the tab traversal order.

We would like to try a touch area on the kick one button like this:

<img width="365" alt="image" src="https://user-images.githubusercontent.com/679486/236012987-ce63db88-7390-4058-a83b-159d6f33c8f8.png">

**Comment by samreid** [comment]
RectangularPushButton doesn't support arbitrary pointer areas:

https://github.com/phetsims/sun/blob/56a5f02524cba687a127894294e9724ce813ee41/js/buttons/RectangularButton.ts#L193-L202

You can dilate and offset.  In the variability screen, it's turning out like this:

<img width="187" alt="image" src="https://user-images.githubusercontent.com/679486/236067314-db6dca26-be3c-4c3a-8884-3435d7ed0ae1.png">

Should we move the kick buttons to be under the kicker?

**Comment by samreid** [comment]
In today's meeting, we think it could be ok to move the kick buttons under the kicker.  But we are also not convinced it is the best design to click on the soccer person to kick the ball.  For instance, in projectile motion, clicking on the cannon does not fire a ball.  What if someone thinks clicking on the person changes the person to the next scene?  We should discuss.

**Comment by samreid** [comment]
In today's meeting, @amanda-phet agreed we are overcomplicating it. The cannon example makes sense.  @catherinecarter tried mocking up the buttons under the kicker, and after consideration, we realized that is how it should be. So we will try the buttons under the kicker, but without changing the pointer area.

**Comment by samreid** [comment]
Fixed. @matthew-blackman can you please review and see if this issue can be closed?

**Comment by marlitas** [comment]
I know this wasn't assigned to me but came up in my notifications so took a quick look.

Wasn't the biggest fan of the magic number adjusting, but it looks like it's probably the best way to move forward. 

I noticed that the offsets for the SoccerPlayNode and the KickButtonGroup do not match, and I'm still not sure why that is the case... it seems their centers would both be the same x value away from 0... perhaps it's a different coordinate frames complication that's not worth looking into. 

KickButtonGroup
```js
centerX: modelViewTransform.modelToViewX( 0 ) - 63,
```
SoccerPlayerNode
```js
this.centerBottom = modelViewTransform.modelToViewPosition( new Vector2( 0, 0 ) ).plusXY( -28, 8.5 );
```

I'm fine to close if @matthew-blackman is.

**Comment by samreid** [comment]
The soccer player graphics are not centered in the "standing still" image, so it is natural that the soccer player would need a custom offset to compensate:

![image](https://github.com/phetsims/center-and-variability/assets/679486/91c9ffec-01ce-409e-ab33-925ee6be1fc3)
![image](https://github.com/phetsims/center-and-variability/assets/679486/611d00a7-cc95-435b-b218-40559faf2520)

**Comment by matthew-blackman** [comment]
I agree with @marlitas comment above regarding magic numbers, but the dynamic player image seems to be a warranted case. Good discussion about the UX here. The button layout looks clean and straightforward. Closing.
